.github/PULL_REQUEST_TEMPLATE/release.md

```diff
@@ -0,0 +1,77 @@
+# Release vFIXME
+
+## Checklist
+
+- [ ] [Create a new github milestone](https://github.com/DataDog/dd-trace-go/milestones/new) for the next version and
+  move any open PRs and issues that are in the current release milestone to the new one.
+
+- [ ] Draft [the release note](https://github.com/DataDog/dd-trace-go/releases/new) and ask the teams for its review on
+  the `#guild-dd-go` channel.
+
+- Deploy a release candidate:
```

Comment:
Can any of this be automated a bit? If we can write a tiny script with these commands below where someone can just provide the version number of the release, that would lower the chances of user error and lower the complexity of the release process overall.
That question applies here and throughout the doc.

Reply:
For now, as a starting point, I automated everything happening in this repository.
Doing this dd-go update automatically would likely involve triggering a separate workflow in dd-go's gitlab CI.
Definitely something to try to do in the future.

So for now I'm only showing here what it should look like to update dd-go's go.mod file.